mlflow/entities/span.py

```diff
@@ -0,0 +1,62 @@
+from __future__ import annotations
+
+import json
+from dataclasses import asdict, dataclass, field
+from typing import Any, Dict, List, Optional, Union
+
+from mlflow.entities._mlflow_object import _MLflowObject
+from mlflow.entities.span_context import SpanContext
+from mlflow.entities.span_event import SpanEvent
+from mlflow.entities.span_status import SpanStatus
+
+
+@dataclass
+class Span(_MLflowObject):
+    """A span object. (TODO: Add conceptual guide for span vs trace.)
+
+    Args:
+        name: Name of the span.
+        context: SpanContext object that contains the request_id (= trace_id) and span_id.
+        parent_span_id: Id of the parent span. If None, the span is the root span.
+        span_type: Type of the span. Can be a pre-defined enum or a custom string.
+        status: Status of the span.
+        start_time: Start time of the span in microseconds.
+        end_time: End time of the span in microseconds.
+        inputs: Inputs data of the span. Optional.
+        outputs: Outputs data of the span. Optional.
+        attributes: Arbitrary key-value pairs of the span attributes. Optional.
+        events: List of events that happened during the span. Optional.
+    """
+
+    name: str
+    context: SpanContext
+    parent_span_id: Optional[str]
+    span_type: Union[SpanType, str]
+    status: SpanStatus
+    start_time: int
```

Comment:
Note: It might be confusing that `timestamp_ms` field in `TraceInfo` is millisecond, while span has microsecond timestamp. However, most likely span will need more precision than millisecond (some code block finishes lightning speed). Alternatively, we could make span's timestamp `float`.

Reply:
The reason for limiting the precision is a downstream restriction on allowable timestamp precision within Spark DataFrames (microseconds are not supported). 
Would it make more sense to keep the default precision within the Trace objects (info, spans) as microseconds and only do the rounding to milliseconds when converting the trace payload for serialization to an inference table? 
That way, we can get the artifact viewing trace ui and the in-notebook visualizations to show accurate durations when it matters most (interactive debugging and development)? 

Thoughts on this @dbczumar ?